apps/console/src/features/applications/components/application-list.tsx

```diff
@@ -298,19 +298,24 @@ export const ApplicationList: FunctionComponent<ApplicationListPropsInterface> =
                                 return template.id === ApplicationManagementConstants.CUSTOM_APPLICATION;
                             }).name;
                     } else {
-                        const templateGroupId: string = applicationTemplates
+                        const relevantApplicationTemplate: ApplicationTemplateListItemInterface | undefined = 
+                            applicationTemplates
                             && applicationTemplates instanceof Array
                             && applicationTemplates.length > 0
                             && applicationTemplates.find((template) => {
                                 return template.id === app.templateId;
-                            }).templateGroup;
-
-                        templateDisplayName = groupedApplicationTemplates
-                            && groupedApplicationTemplates instanceof Array
-                            && groupedApplicationTemplates.length > 0
-                            && groupedApplicationTemplates.find((group) => {
-                                return (group.id === templateGroupId || group.templateGroup === templateGroupId);
-                            }).name;
+                            });
+
+                        if (relevantApplicationTemplate?.templateGroup) {
+                            const templateGroupId: string = relevantApplicationTemplate.templateGroup;
```

Comment:
Can we check for null using `?`. Check the other places also

Reply:
Used the `?` in the first place to address this comment https://github.com/wso2/identity-apps/pull/3463#discussion_r973876598.
And the checking whether the templateGroup is null by using the `if` statement.